lib/ansible/modules/packaging/language/pip.py

```diff
@@ -271,15 +273,22 @@ def _get_packages(module, pip, chdir):
     return (command, out, err)
 
 
-def _is_present(name, version, installed_pkgs, pkg_command):
+def _is_present(module, name, version, installed_pkgs, pkg_command):
     '''Return whether or not package is installed.'''
+    if version is not None:
+        if version[0].isdigit():
+            version = "==%s" % version
+        try:
+            vp = VersionPredicate("%s (%s)" % (name, version))
+        except ValueError:
+            module.fail_json(msg="Can parse invalid version number %s" % version)
     for pkg in installed_pkgs:
         if '==' in pkg:
             pkg_name, pkg_version = pkg.split('==')
         else:
             continue
 
-        if pkg_name == name and (version is None or version == pkg_version):
+        if (pkg_name.lower() == name.lower()) and (version is None or vp.satisfied_by(pkg_version)):
```

Comment:
Why is the `.lower()` needed here?

Reply:
A corner case I found in my machine like user want to `install django`, but the ```pip list``` return ```Django```. Pip is case insensitive but some package in ```pip list``` have capital name.

Reply:
If you'll have a class for distribution, you'll be able to encapsulate lot's of implementation details there and even keep a normalized package name there internally.